### DISCUSSION TITLE
controlling inheritance between auxkernels

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#16996] [Q&A General] [answered]
Dear all,
I was sure my question have been already asked (therefore I do apologize for the multiple posting), but I could not find anything similar in the discussion forum ...
My question: Is there any way to control the order of inheritance among different auxkernels/variables?
Problem: I have an auxkernel (aux_1) that computes the difference between the value of an auxvariable over time (u-u_old) where the auxvariable is provided by a different auxkernel (aux_0). This said, I need to force the computation of the latter (aux_0) to be done before I am calling it inside aux_1. 
The main issue I see is that both kernels must be computed  at timestep_end since aux_0 requires the values of a material property and I want to pass the final value of the same quantity. I know I could do the same computation at a material level, but still I would love it to be done at the auxkernel level if possible. 
Any help would be highly appreciated,
Mauro

**Comment by permcody** [comment, marked as answer]
You mean "order of execution". MOOSE has a basic "dependency resolution" system built in to compute things that it knows about in the right order. As long as your `execute_on` matches and you use normal coupling syntax to link to variables that you depend on, objects will be computed in the right order. You can find several tests in our suite that demonstrate that execution order is handled:
[https://github.com/idaholab/moose/tree/next/test/tests/auxkernels/element_aux_var](https://github.com/idaholab/moose/tree/next/test/tests/auxkernels/element_aux_var)

**Reply by mcacace** [reply]
Thanks, and of course "execution order" is the right word. Will test my auxkernels and let you know in case I have any problems.
Thanks again,
Mauro
